chapters/request_size.rst

```diff
@@ -0,0 +1,133 @@
+codes as example.
+
+We first import the necessary modules.
+
+.. highlight:: python
+.. code-block:: python
+
+    import time
+    from typing import Iterator
+
+    import numpy as np
+    from jina import Document
+    from jina.executors.crafters import BaseCrafter
+    from jina.flow import Flow
+
+Then we define a simple crafter which will just forward the data.
+
+.. highlight:: python
+.. code-block:: python
+
+    class SimpleCrafter(BaseCrafter):
+        def craft(self, id, *args, **kwargs):
+            return {'id': id}
+
+
+Next we define the data generator.
+
+.. highlight:: python
+.. code-block:: python
+
+    def random_docs(num_docs, chunks_per_doc=5, embed_dim=10, jitter=1, start_id=0, embedding=True) -> Iterator['Document']:
+        next_chunk_doc_id = start_id + num_docs
+        for j in range(num_docs):
+            doc_id = start_id + j
+
+            d = Document(id=doc_id)
+            d.text = b'hello world'
+            d.tags['id'] = doc_id
+            if embedding:
+                d.embedding = np.random.random([embed_dim + np.random.randint(0, jitter)])
+            d.update_content_hash()
+
+            for _ in range(chunks_per_doc):
+                chunk_doc_id = next_chunk_doc_id
+
+                c = Document(id=chunk_doc_id)
+                c.text = 'i\'m chunk %d from doc %d' % (chunk_doc_id, doc_id)
+                if embedding:
+                    c.embedding = np.random.random([embed_dim + np.random.randint(0, jitter)])
+                c.tags['parent_id'] = doc_id
+                c.tags['id'] = chunk_doc_id
+                c.update_content_hash()
+                d.chunks.append(c)
+                next_chunk_doc_id += 1
+
+            yield d
+
+
+For this example, we will index 100 documents and use 10 parallel crafters. The ``request_size``
+is set as 20. So the 100 documents will be divided into 5 parts and then distributed to the crafters.
+
+.. highlight:: python
+.. code-block:: python
+
+    def main():
+
+        request_size=20
+        start_time = time.time()
+
+        f = Flow(runtime='process').add(
+            name='simple_crafter',
+            uses='SimpleCrafter',
+            parallel=10)
+        with f:
+            f.index(input_fn=random_docs(100), request_size=request_size)
+
+        end_time = time.time()
+
+        seconds_elapsed = end_time - start_time
+        print(seconds_elapsed)
+
+
+    if __name__ == '__main__':
+        main()
+
+
+Choose different request size
+--------------------
+Different setting of ``request_size`` may influence the running performance.
+A higher value means large size data will be fed into the ``Pea`` and will demand more memory.
+A lower value will decrease the cost of memory but may increase the running time.
+
+An simple extension of the above example generate a box plot showing the relationship between ``request_size`` and running time
```

Comment:
do we have this extension?

Reply:
No, we don't. That just means we extend the code example above by setting different request size values to help users get more insights on the influence of this parameter. Not sure if this is a good 😄